src/java.base/share/native/libjava/ClassLoader.c

```diff
@@ -88,28 +89,32 @@ Java_java_lang_ClassLoader_defineClass1(JNIEnv *env,
 
     if (data == NULL) {
         JNU_ThrowNullPointerException(env, 0);
-        return 0;
+        return NULL;
     }
 
     /* Work around 4153825. malloc crashes on Solaris when passed a
      * negative size.
      */
     if (length < 0) {
         JNU_ThrowArrayIndexOutOfBoundsException(env, 0);
-        return 0;
+        return NULL;
     }
 
-    body = (jbyte *)malloc(length);
+    // On malloc(0), implementators of malloc(3) have the choice to return either
+    // NULL or a unique non-NULL pointer. To unify libc behavior across our platforms
+    // we chose the latter. (see 8283225)
+    body = (jbyte *)malloc(length < 1 ? 1 : length);
```

Comment:
This code conflates a length == in the comment with length < 1 in the code.
If the issue is with length == 0, make that be the test.

Reply:
Thanks for your comment. I agree, and will make this change.